src/programs/city/construct.js

```diff
@@ -52,6 +52,10 @@ class CityConstruct extends kernel.process {
           continue
         }
 
+        if (structure.hits === undefined) { // Skips things that can't be removed, I.E. Novice area walls, SK lair (can break sim)
```

Comment:
This `if` can actually replace the controller check in line 40. I'd also move it up (i.e. to line 40) because it's a very cheap check that can be done before calling `getStructureAt()`.

Reply:
Done. Nice comment, thanks.